File: mobile/lib/features/expenses/data/expense_model.dart
class ExpenseModel {
  final String id;
  final String category;
  final double amount;
  final DateTime date;
  final String note;
  final Map<String, dynamic> recordedBy;

  ExpenseModel({
    required this.id,
    required this.category,
    required this.amount,
    required this.date,
    required this.note,
    required this.recordedBy,
  });

  factory ExpenseModel.fromJson(Map<String, dynamic> json) {
    return ExpenseModel(
      id: json['_id'] ?? json['id'] ?? '',
      category: json['category'] ?? '',
      amount: (json['amount'] as num?)?.toDouble() ?? 0.0,
      date: DateTime.parse(json['date']),
      note: json['note'] ?? '',
      recordedBy: json['recordedBy'] ?? {},
    );
  }
}

File: server/src/modules/expenses/expense.model.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IExpense extends Document {
  category: string;
  amount: number;
  date: Date;
  note: string;
  recordedBy: mongoose.Types.ObjectId;
  createdAt: Date;
  updatedAt: Date;
}

const expenseSchema = new Schema<IExpense>(
  {
    category: { type: String, required: true },
    amount: { type: Number, required: true, min: 0 },
    date: { type: Date, required: true },
    note: { type: String, required: true },
    recordedBy: { type: Schema.Types.ObjectId, ref: 'User', required: true },
  },
  { timestamps: true }
);

export const ExpenseModel = mongoose.model<IExpense>('Expense', expenseSchema);
